src/main/java/com/example/bytex/user/UserEntityServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.example.bytex.user;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class UserEntityServiceImpl implements UserEntityService {
+
+    UserEntityRepository userEntityRepository;
+
+    @Autowired
+    public UserEntityServiceImpl(UserEntityRepository userEntityRepository) {
+        this.userEntityRepository = userEntityRepository;
+    }
+
+    @Override
+    public List<UserEntity> getUsers() {
+        return userEntityRepository.findAll();
+
+    }
+
+    @Override
+    public void addNewUser(UserEntity userEntity) {
+        userEntityRepository.save(userEntity);
+    }
+
+    @Override
+    public void deleteUser(Long id) {
+        boolean existsInDatabase = userEntityRepository.existsById(id);
+
+        if(existsInDatabase)
+        {
+            userEntityRepository.deleteById(id);
+        }
+        else
+        {
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "User with id "+ id + " doesnt exist");
```

Comment:
You should take into account separating the `controller` layer and the `service` layer. With this approach, from the `service` layer you are throwing an exception that makes sense for a `controller`, but if you were to use the same service in another service (e.g. `UserProfileService` which needs to get an user by id: `UserEntityService.findById(..)`), then the `ResponseStatusException` does not make sense